components/nimbus/src/stateful/evaluator.rs

```diff
@@ -47,6 +51,10 @@ impl From<AppContext> for TargetingAttributes {
 }
 
 impl TargetingAttributes {
+    pub(crate) fn with_recorded_context(&mut self, recorded_context: Arc<dyn RecordedContext>) {
+        self.recorded_context = Some(recorded_context.to_json());
+    }
```

Comment:
Nit: `with_xyz` methods are usually a builder-esque pattern so I would expect this to return a `TargetingAttribtues` or `&mut TargetingAttributes`. Instead, maybe this should be called `record_context()` ?

Reply:
what about `set_recorded_context`?

Reply:
:+1: